docs/source/advanced.rst

```diff
@@ -258,23 +258,217 @@ of :py:class:`scikeras.wrappers.BaseWrappers`. Currently, they are:
+
+    clf = KerasClassifier(
+        ...
+        metrics=BinaryAccuracy,
+        metrics__threashold=0.65,
+    )
+    # or to apply multiple metrics to all outputs
+    clf = KerasClassifier(
+        ...
+        metrics=[BinaryAccuracy, AUC],  # both applied to all outputs
+        metrics__0__threashold=0.65,
+    )
+    # or for different metrics to each output
+    clf = KerasClassifier(
+        ...
+        metrics={"out1": BinaryAccuracy, "out2": [BinaryAccuracy, AUC]},
+        metrics__out1__threashold=0.65,
+        metrics__out2__0__threashold=0.65,
+    )
+    # assuming you ordered your outputs like [out1, out2], this is equivalent to
+    clf = KerasClassifier(
+        ...
+        metrics=[BinaryAccuracy, [AUC, BinaryAccuracy]],
+        metrics__0__threashold=0.65,
+        metrics__1__0__threashold=0.65,
+    )
+
+
+See the `Keras Metrics docs`_ for more details on mapping metrics to outputs.
+
+Callbacks
++++++++++
+
+.. code:: python
+
+    clf = KerasClassifier(
+        ...
+        callbacks=tf.keras.callbacks.EarlyStopping
+        callbacks__monitor="loss",
+    )
+
+Just like metrics and losses, callbacks support several syntaxes to compile them depending on your needs:
+
+.. code:: python
+
+    # for multiple callbacks using dict syntax
+    clf = KerasClassifier(
+        ...
+        callbacks={"bl": tf.keras.callbacks.BaseLogger, "es": tf.keras.callbacks.EarlyStopping}
+        callbacks__es__monitor="loss",
+    )
+    # or using list sytnax
+    clf = KerasClassifier(
+        ...
+        callbacks=[tf.keras.callbacks.BaseLogger, tf.keras.callbacks.EarlyStopping]
+        callbacks__1__monitor="loss",  # EarlyStopping(monitor="loss")
+    )
+
+Keras callbacks are event based, and are triggered depending on the methods they implement.
+For example:
+
+.. code:: python
+    from tensorflow import keras
+
+    class MyCallback(keras.callbacks.Callback):
+
+        def on_train_begin(self, epoch, logs=None):
+            print("Started training from a `fit` or `partial_fit` call!")
+
+        def on_test_begin(self, epoch, logs=None):
+            print("Started testing/evaluation from a `fit` call!")
+
+        def on_predict_begin(self, epoch, logs=None):
+            print("Started prediction from a `predict` or `predict_proba` call!")
+
+See the `Keras Callbacks docs`_ for more details on method-based event dispatch.
+
+If for any reason you do need to create seperate callbacks for ``fit`` and ``predict``,
+simply use the ``fit__`` or ``predict__`` routing prefixes on your callback:
+
+    clf = KerasClassifier(
+        ...
+        callbacks=tf.keras.callbacks.Callback,  # called from both fit and predict
+        fit__callbacks=tf.keras.callbacks.Callback,  # called only from fit
+        predict__callbacks=tf.keras.callbacks.Callback,  # called only from predict
+    )
+
+Any routed constructor parameters must also use the corresponding prefix to get routed correctly.
+
+Routing as args or kwargs
```

Comment:
```suggestion
Routing as arguments/keyword arguments
```

"args" and "kwargs" are technical jargon.

Reply:
Thank you, incorporated.